sql/sql_acl.cc

```diff
@@ -13087,11 +13117,34 @@ bool Sql_cmd_grant_proxy::execute(THD *thd)
     return true;
 
   WSREP_TO_ISOLATION_BEGIN(WSREP_MYSQL_DB, NULL, NULL);
-  /* Conditionally writes to binlog */
-  if (mysql_grant(thd, NULL/*db*/, lex->users_list, {m_grant_option, is_revoke()},
-                  true/*proxy*/))
+
+  if (tables.open_and_lock(thd, Table_user | Table_proxies_priv, TL_WRITE))
     return true;
```

Comment:
old code had here
```c
   if ((result= tables.open_and_lock(thd, tables_to_open, TL_WRITE)))
     DBUG_RETURN(result != 1);
```
now you return an error when `result == 1`, this ignores the logic from the comment
```
     @retval  1 replication filters matched. Abort the operation,
                but return OK (!)
```